src/ucp/core/ucp_request.c

```diff
@@ -357,9 +357,13 @@ ucp_request_send_start(ucp_request_t *req, ssize_t max_short,
         if (length <= (msg_config->max_bcopy - proto->only_hdr_size)) {
             req->send.uct.func = proto->bcopy_single;
             UCS_PROFILE_REQUEST_EVENT(req, "start_bcopy_single", req->send.length);
-        } else {
+        } else  if (proto->bcopy_multi != NULL) {
```

Comment:
one less space

Reply:
also, can we check rndv_thresh here? and assert(proto->bcopy_multi != NULL) if less than rndv thresh

Reply:
no, because rndv_thresh may be larger than this size, because of higher max_zcopy thresh. But we can't use zcopy for genetic types, and we can't use bcopy-multi for tag offload

Reply:
but it seems like config->tag.rndv_send_nbr.am_thresh will be set to lower size after this PR, because of https://github.com/openucx/ucx/pull/5151/files#diff-8fd5f37669b4533d1e9389ff6839dcccR1592?